runtime/lua/vim/lsp/handlers.lua

```diff
@@ -16,68 +16,76 @@ local function err_message(...)
   api.nvim_command('redraw')
 end
 
---see: https://microsoft.github.io/language-server-protocol/specifications/specification-current/#workspace_executeCommand
+-- https://microsoft.github.io/language-server-protocol/specifications/specification-current/#workspace_executeCommand
+---@type vim.lsp.ResponseHandler
 M[ms.workspace_executeCommand] = function(_, _, _, _)
   -- Error handling is done implicitly by wrapping all handlers; see end of this file
 end
 
---see: https://microsoft.github.io/language-server-protocol/specifications/specification-current/#progress
----@param result lsp.ProgressParams
----@param ctx lsp.HandlerContext
-M[ms.dollar_progress] = function(_, result, ctx)
+-- https://microsoft.github.io/language-server-protocol/specifications/specification-current/#progress
+---@param params lsp.ProgressParams
+---@type vim.lsp.NotificationHandler
+M[ms.dollar_progress] = function(_, params, ctx)
   local client = vim.lsp.get_client_by_id(ctx.client_id)
   if not client then
     err_message('LSP[id=', tostring(ctx.client_id), '] client has shut down during progress update')
+    -- TODO(wookayin): This is a notification handler so should be removed (after testing!) #16472
     return vim.NIL
   end
   local kind = nil
-  local value = result.value
+  local value = params.value
 
   if type(value) == 'table' then
+    ---@cast value lsp.WorkDoneProgressBegin|lsp.WorkDoneProgressReport|lsp.WorkDoneProgressEnd
     kind = value.kind
     -- Carry over title of `begin` messages to `report` and `end` messages
     -- So that consumers always have it available, even if they consume a
     -- subset of the full sequence
     if kind == 'begin' then
-      client.progress.pending[result.token] = value.title
+      client.progress.pending[params.token] = value.title
     else
-      value.title = client.progress.pending[result.token]
+      value.title = client.progress.pending[params.token]
       if kind == 'end' then
-        client.progress.pending[result.token] = nil
+        client.progress.pending[params.token] = nil
       end
     end
   end
 
-  client.progress:push(result)
+  client.progress:push(params)
 
   api.nvim_exec_autocmds('LspProgress', {
     pattern = kind,
     modeline = false,
-    data = { client_id = ctx.client_id, result = result },
+    -- TODO(wookayin): breaking change during 0.10-nightly, results -> params. Revise data again
+    data = { client_id = ctx.client_id, result = params },
```

Comment:
Note this --- we probably had a wrong name for `LspProgress`. #23958 

A breaking change (during 0.10-dev only) will be needed.